### PR TITLE
Add basic support for PocketMine-MP

### PR DESCRIPTION
**Pull request by Mstrodl**
Title says all. Source for the "PufferMine.jar" file can be found [here](https://github.com/MStrodl/PufferMine). Synopsis of installer:

-b base path (/home)
-u user to run it (root)
-m PocketMine-MP release channel (stable) (can be: development/beta/stable)
-p plugin (pocketmine)

As you can see the only new option is -m which shows the channel and by default is stable

### CODE REVIEW

**Comment by DDynamic** [comment]
Looks great! However, please note a few things
**1.** When/If this gets merged, you might need to pull request the panel to include the new gamemode. (Not sure about this, but I think so.)
**2.** @DaneEveritt the main PufferPanel/Scales developer, is AFK during the summer, so it may be a month or two before this gets merged. 

However, I thank you for your contribution!

**Comment by Mstrodl** [comment]
@BlameDylan Thanks! I was thinking the same as well, do you have any idea how to do this? If not, that's fine I'll try and figure it out myself :smiley:

**EDIT:** Never mind I figured it out and have made another pull request :grinning:

**Comment by Omeryl** [comment]
Any particular reason you didn't smash your commits before submitting a PR? It would've been nicer to be honest.

**Comment by Mstrodl** [comment]
What do you mean @Omeryl ? What does it mean to smash a commit (and how do I do so?)?

**Comment by puffrfish** [comment]
@BlameDylan There are other developers who can merge than just Dane.

**Comment by Omeryl** [comment]
@Mstrodl For future reference, you're flooding the tree with a ton of commits with (useless) emoji. Read: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

EDIT: This is my personal opinion, I'm not a core dev.

**Comment by puffrfish** [comment]
Sorry, this isn't an acceptable way of doing this. Servers should be run natively by the plugin, not with a "trick" jar.

**Comment by DDynamic** [comment]
@puffrfish, Sorry, I thought you all had access to just the PufferPanel repo. My bad!

File: lib/plugins/pocketmine.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+settings = {
+	name: "Pocketmine-MP",
+	stop: "stop",
+	exe: "java",
+	jar: "data/plugins/pocketmine/QuietPufferMine.jar",
+	cfg: "server.properties",
+	trigger: {
+		started: ')! For help, type "help" or "?"',
+		// eula: 'Go to eula.txt for more info.'
```
Comment (DaneEveritt) [inline lib/plugins/pocketmine.js:22]: If you are going to do this you need to remove the `,` on the previous line.

Comment (Omeryl) [inline lib/plugins/pocketmine.js:22]: @DaneEveritt For what it's worth: trailing commas work fine in JavaScript like that. 

``` javascript
root@vinteum:~# node
> t = {a: 1, b:2, }
{ a: 1, b: 2 }
>
```

**Comment by Mstrodl** [comment]
O.K. I'm working on a better fix for this sorry I caused any controversy. I also have a few other plugins I'd like to share in the future.

**Pull request closed without merging**